lib/wallet/account/hot-wallet-account.ts

```diff
@@ -34,7 +40,191 @@ export class HotWalletAccount implements IWallet {
     private readonly networkId: NetworkType,
   ) {}
 
-  public getPrivateKey() {
+  async performCommitReveal(
+    payload: any,
+    revealPriorityFee: IGeneratorSettingsObject["priorityFee"],
+    extraOutputs: IPaymentOutput[] = [],
+  ) {
+    const privateKey = this.getPrivateKey();
+    const publicKey = privateKey.toPublicKey();
+    const address = publicKey.toAddress(this.networkId);
+
+    let eventReceived = false;
+    let addedEventTxId: any;
+    let submittedTxId: any;
+
+    const handleUtxosChanged = async (event: any) => {
+      const removedEntry = event.data.removed.find(
+        (entry: any) => entry.address.payload === address.payload,
+      );
+      const addedEntry = event.data.added.find(
+        (entry: any) => entry.address.payload === address.payload,
+      );
+
+      addedEventTxId =
+        addedEntry?.outpoint?.transactionId ??
+        removedEntry?.outpoint?.transactionId;
+
+      if (addedEventTxId == submittedTxId) {
+        eventReceived = true;
+      }
+    };
+
+    await this.rpcClient.subscribeUtxosChanged([address.toString()]);
+    this.rpcClient.addEventListener("utxos-changed", handleUtxosChanged);
+
+    const script = new ScriptBuilder()
+      .addData(publicKey.toXOnlyPublicKey().toString())
+      .addOp(Opcodes.OpCheckSig)
+      .addOp(Opcodes.OpFalse)
+      .addOp(Opcodes.OpIf)
+      .addData(Buffer.from("kasplex"))
+      .addI64(0n)
+      .addData(Buffer.from(JSON.stringify(payload, null, 0)))
+      .addOp(Opcodes.OpEndIf);
+
+    const P2SHAddress = addressFromScriptPublicKey(
+      script.createPayToScriptHashScript(),
+      this.networkId,
+    );
+
+    if (!P2SHAddress) {
+      throw new Error("Invalid P2SH address");
+    }
+
+    const { entries } = await this.rpcClient.getUtxosByAddresses({
+      addresses: [address.toString()],
+    });
+    const { transactions } = await createTransactions({
```

Comment:
It seems make more sense to use `createTransaction` to create one transaction instead of multiple.